tests/test_dle.py

```diff
@@ -334,14 +335,18 @@ def test_cache_blocking_imperfect_nest_v2(blockinner):
     op2 = Operator(eq, opt=('advanced-fsg', {'blockinner': blockinner}))
 
     # First, check the generated code
-    trees = retrieve_iteration_tree(op2._func_table['bf0'].root)
+    # Rule out len(i) <= 1 trees
+    trees = [i for i in retrieve_iteration_tree(op2) if len(i) > 1]
+
     assert len(trees) == 2
     assert len(trees[0]) == len(trees[1])
-    assert all(i is j for i, j in zip(trees[0][:2], trees[1][:2]))
-    assert trees[0][2] is not trees[1][2]
-    assert trees[0].root.dim.is_Incr
-    assert trees[1].root.dim.is_Incr
-    assert op2.parameters[6] is trees[0].root.step
+    assert all(i is j for i, j in zip(trees[0][:3], trees[1][:3]))
+    assert trees[0][2] is trees[1][2]
+    assert trees[0][3] is not trees[1][3]
```

Comment:
I hope all of these checks are as strong as in master...

Reply:
Should be good I think